File: DataExtractionOSM/src/net/osmand/ResultMatcher.java
package net.osmand;

/**
 * Easy matcher to be able to publish results immediately
 * 
 */
public interface ResultMatcher<T> {

	/**
	 * @param name
	 * @return true if result should be added to final list
	 */
	boolean publish(T object);
	
	/**
	 * @returns true to stop processing
	 */
	boolean isCancelled();

}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/binary/ResultMatcher.kt
package net.osmand.shared.binary

/**
 * Where a search hands its results as it finds them, and how it learns it should stop.
 *
 * A copy of `net.osmand.ResultMatcher`, which stays in OsmAnd-java.
 */
interface ResultMatcher<T> {

	/** @return true if the [obj] is accepted, false otherwise. */
	fun publish(obj: T): Boolean

	/** @return true to stop the search. */
	fun isCancelled(): Boolean
}
